.github/workflows/codecov_unittest.yaml

```diff
@@ -0,0 +1,25 @@
+name: Unit Test And Code Coverage
+
+# this workflow would work on all the branches and all the prs
+on:
+  push: {}
+  pull_request: {}
+
+jobs:
+  unitTestAndCodeCoverage:
+    name: "Unit Test And Code Coverage"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
Oops, I just copy-paste that from other PR, let me take a look.

Reply:
As the document says: " When checking out the repository that triggered a workflow, this defaults to the reference or SHA for that event." So I think we could just remove this parameter, it could play well with both `pull_request` and `push` by itself. :D

Reply:
Updated PTAL :)